libs/vkd3d/command.c

```diff
@@ -6557,69 +6653,113 @@ static void STDMETHODCALLTYPE d3d12_command_list_SetPredication(d3d12_command_li
 {
     struct d3d12_command_list *list = impl_from_ID3D12GraphicsCommandList(iface);
     struct d3d12_resource *resource = unsafe_impl_from_ID3D12Resource(buffer);
-    const struct vkd3d_vulkan_info *vk_info = &list->device->vk_info;
-    const struct vkd3d_vk_device_procs *vk_procs;
+    const struct vkd3d_vk_device_procs *vk_procs = &list->device->vk_procs;
+    const struct vkd3d_predicate_ops *predicate_ops = &list->device->meta_ops.predicate;
+    struct vkd3d_predicate_resolve_args resolve_args;
+    VkConditionalRenderingBeginInfoEXT begin_info;
+    VkPipelineStageFlags dst_stages, src_stages;
+    struct vkd3d_scratch_allocation scratch;
+    VkAccessFlags dst_access, src_access;
+    VkMemoryBarrier vk_barrier;
+    VkBufferCopy copy_region;
 
     TRACE("iface %p, buffer %p, aligned_buffer_offset %#"PRIx64", operation %#x.\n",
             iface, buffer, aligned_buffer_offset, operation);
 
-    if (!vk_info->EXT_conditional_rendering)
+    d3d12_command_list_end_current_render_pass(list, true);
+
+    if (resource && (aligned_buffer_offset & 0x7))
+        return;
+
+    if (!list->device->device_info.buffer_device_address_features.bufferDeviceAddress &&
```

Comment:
Shouldn't this be ||?

Reply:
Or hm, && is probably correct since there are multiple ways to deal with this.

Reply:
Yes, we only want to error out if *both* features are unsupported. We only require BDA for the fallback path, and for 64-bit predicates to work correctly.

Reply:
Well then.